integration/tests/epochs/cohort2/epoch_recover_from_efm_test.go

```diff
@@ -165,13 +175,103 @@ func (s *RecoverEpochSuite) TestRecoverEpoch() {
 	)
 
 	// 3. Submit recover epoch transaction to the network.
-	// submit the recover epoch transaction
 	env := utils.LocalnetEnv()
 	result := s.recoverEpoch(env, txArgs)
 	require.NoError(s.T(), result.Error)
 	require.Equal(s.T(), result.Status, sdk.TransactionStatusSealed)
 
-	// 3. Ensure EpochRecover event was emitted.
+	// 4. Ensure expected EpochRecover event is emitted.
+	eventType := ""
+	for _, evt := range result.Events {
+		if strings.Contains(evt.Type, "FlowEpoch.EpochRecover") {
+			eventType = evt.Type
+			break
+		}
+	}
+	require.NotEmpty(s.T(), eventType, "expected FlowEpoch.EpochRecover event type")
+	events, err := s.Client.GetEventsForBlockIDs(s.Ctx, eventType, []sdk.Identifier{result.BlockID})
+	require.NoError(s.T(), err)
+	require.Equal(s.T(), events[0].Events[0].Type, eventType)
+
+	// 5. Ensure the network transitions into the recovery epoch and finalizes the first view of the recovery epoch.
+	startViewOfNextEpoch := uint64(txArgs[1].(cadence.UInt64))
+	s.TimedLogf("waiting to transition into recovery epoch (finalized view %d)", startViewOfNextEpoch)
+	s.AwaitFinalizedView(s.Ctx, startViewOfNextEpoch, 2*time.Minute, 500*time.Millisecond)
+	s.TimedLogf("observed finalized first view of recovery epoch %d", startViewOfNextEpoch)
+
+	s.AssertInEpoch(s.Ctx, 1)
+}
+
+// TestRecoverEpochNodeEjected ensures that the recover epoch governance transaction flow works as expected, and a network that
+// enters Epoch Fallback Mode can successfully recover.
+// For this specific scenario, we are testing a scenario where the consensus committee is a subset of the DKG committee, i.e.,
+// a node was ejected between epoch start and submitting the recover epoch transaction.
+// This test will do the following:
+// 1. Triggers EFM by turning off the sole collection node before the end of the DKG forcing the DKG to fail.
+// 2. Generates epoch recover transaction args using the epoch efm-recover-tx-args.
+// 3. Eject consensus node by modifying the snapshot before generating the recover epoch transaction args.
+// 4. Submit recover epoch transaction.
+// 5. Ensure expected EpochRecover event is emitted.
+// 6. Ensure the network transitions into the recovery epoch and finalizes the first view of the recovery epoch.
+func (s *RecoverEpochSuite) TestRecoverEpochNodeEjected() {
+	// 1. Manually trigger EFM
+	// wait until the epoch setup phase to force network into EFM
+	s.AwaitEpochPhase(s.Ctx, 0, flow.EpochPhaseSetup, 10*time.Second, 500*time.Millisecond)
+
```

Comment:
```suggestion
```
In principle, stopping the Collection Node _before_ entering the EpochSetup phase would be more reliable. The Consensus Nodes send off the first DKG message as soon as the EpochSetup phase starts, and can technically send their result submission after the DKG final view, so this could be race-prone.